crates/uv-toolchain/src/toolchain.rs

```diff
@@ -43,66 +44,53 @@ impl Toolchain {
     /// See [`uv-toolchain::discovery`] for implementation details.
     pub fn find(
         python: Option<ToolchainRequest>,
-        system: SystemPython,
-        preview: PreviewMode,
+        environments: EnvironmentPreference,
+        preference: ToolchainPreference,
         cache: &Cache,
     ) -> Result<Self, Error> {
-        if let Some(request) = python {
-            Self::find_requested(&request, system, preview, cache)
-        } else if system.is_preferred() {
-            Self::find_default(preview, cache)
-        } else {
-            // First check for a parent interpreter
-            // We gate this check to avoid an extra log message when it is not set
-            if std::env::var_os("UV_INTERNAL__PARENT_INTERPRETER").is_some() {
-                match Self::find_parent_interpreter(system, cache) {
-                    Ok(env) => return Ok(env),
-                    Err(Error::NotFound(_)) => {}
-                    Err(err) => return Err(err),
-                }
-            }
-
-            // Then a virtual environment
-            match Self::find_virtualenv(cache) {
-                Ok(venv) => Ok(venv),
-                Err(Error::NotFound(_)) if system.is_allowed() => {
-                    Self::find_default(preview, cache)
-                }
-                Err(err) => Err(err),
-            }
-        }
+        let request = python.unwrap_or_default();
+        Self::find_requested(&request, environments, preference, cache)
```

Comment:
This is a big plus. We don't need to do multiple discovery attempts to get a different ordering. We just say what we want.

Reply:
I like it.